libparsec/crates/types/tests/id.rs

```diff
@@ -15,31 +15,43 @@ fn device_label_bad_size() {
 }
 
 #[rstest]
-#[case("foo42")]
-#[case("FOO")]
-#[case("f")]
-#[case("f-o-o")]
-#[case("f_o_o")]
-#[case(&"x".repeat(32))]
-#[case("三国")]
+#[case::text_32_long("This_one_is_exactly_32_char_long")]
+#[case::alphanum("01239AbcdDEFA")]
+#[case::with_underscore("hello_World")]
+#[case::with_hyphen("hello-World")]
+#[case::text_32_long("This_one_is_exactly_32_char_long")]
+// cspell:disable-next-line
+#[case::with_unicode_1("ªµºÖØøˆˠˬˮ\u{301}ͶͽͿΆΊ")]
+#[case::with_unicode_2("ΌΎΣҁ\u{484}Աՙֈ\u{5af}\u{5bf}\u{5c1}\u{5c4}\u{5c7}")]
+#[case::with_unicode_3("ת\u{5ef}\u{610}٩ۓە")]
+#[case::with_unicode_4("\u{6df}\u{6ea}ۿܐݍ߀ߺ\u{7fd}ࠀࡀ")]
+// cspell:disable-next-line
+#[case::hello("𝔅𝔬𝔫𝔧𝔬𝔲𝔯")]
 fn organization_id_user_id_and_device_name(#[case] raw: &str) {
-    let organization_id = OrganizationID::from_str(raw).unwrap();
+    use unicode_normalization::UnicodeNormalization;
+
+    let raw = raw.nfc().collect::<String>();
```

Comment:
I'd rather do a sanity check on the test input that convert them (this way the test stop if something is unexpected instead of running with unexpected data)

```suggestion
    // Sanity check to avoid debugging visually same but different stuff
    let nfc_raw = raw.nfc().collect::<String>();
    p_assert_eq!(nfc_raw, raw);
```

Reply:
If I've done that, it is because the `raw != nfc_raw`

Reply:
if `raw != nfc_raw` that means our test is broken

Typically is `ΌΎΣҁ\u{484}Աՙֈ\u{5af}\u{5bf}\u{5c1}\u{5c4}\u{5c7}` NFC ? We don't know. Yet this is the string we will look it for debugging. So we will have to run the nfc conversion to then have the correct string value.

So I think it is better to provided-valid-value-and-assert-they-are-what-we-expect than to provide-potentially-invalid-value-then-convert-it-into-something-valid

Reply:
so you know, by adding `p_assert_eq!(raw.as_bytes(), raw.nfc().collect::<String>().as_bytes());` the test fail. So indeed some of the test data are not nfc :stuck_out_tongue:

Reply:
done